### PR TITLE
[receiver/otlp] Promote `component.UseLocalHostAsDefaultHost` to beta

### PR DESCRIPTION
**Pull request by mx-psi**
#### Description

Promotes `component.UseLocalHostAsDefaultHost` feature gate to beta.

#### Link to tracking issue

Updates #8510

### CODE REVIEW

**Comment by yurishkuro** [comment]
+1

**Review by atoulme — APPROVED** [review]
+1

File: .chloggen/mx-psi_enable-component-localhost-fgh.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: breaking
+
+# The name of the component, or a single word describing the area of concern, (e.g. otlpreceiver)
+component: otlpreceiver
```
Comment (TylerHelmuth) [inline .chloggen/mx-psi_enable-component-localhost-fgh.yaml:7]: This affects more than the otlp receiver right? Any receivers using `ToServer` is affected

Comment (mx-psi) [reply .chloggen/mx-psi_enable-component-localhost-fgh.yaml:7]: I'll add an extra API release note on here, and list the components explicitly on the contrib changelog. Does that make sense?

File: otelcol/testdata/otel-log-to-file.yaml [hunk 2]
```diff
@@ -14,7 +14,7 @@ exporters:
 service:
   telemetry:
     logs:
-      level: warn
+      level: info
```
Comment (mx-psi) [inline otelcol/testdata/otel-log-to-file.yaml:17]: This is to make the Windows service test pass. Before this change, we had the warning from confighttp and configgrpc, but after it there are no logs at warn level with this config, which makes the assertion that the logs were stored in a file fail

**Comment by mx-psi** [comment]
Please also take a look at the contrib counterpart, open-telemetry/opentelemetry-collector-contrib/pull/33658, cc @TylerHelmuth

File: .chloggen/mx-psi_enable-component-localhost-fgh.yaml (hunk 1 above)
Comment (mx-psi) [reply .chloggen/mx-psi_enable-component-localhost-fgh.yaml:7]: Actually, I don't think an API release note is needed here, this does not change anything in confighttp/configrpc, only in the specific components

**Review by songy23 — APPROVED** [review]

**Review by djaglowski — APPROVED** [review]

**Review by TylerHelmuth — APPROVED** [review]

**Review by evan-bradley — APPROVED** [review]

**Pull request merged**